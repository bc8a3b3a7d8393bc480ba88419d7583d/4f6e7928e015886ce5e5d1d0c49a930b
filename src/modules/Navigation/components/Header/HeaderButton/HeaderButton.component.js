import React from 'react';
import PropTypes from 'prop-types';
import { View, TouchableOpacity } from 'react-native';
import { withNavigation } from 'react-navigation';

import styles from './HeaderButton.component.styles';

const HeaderButton = ({ children, isRightButton, onPress }) => (
  <TouchableOpacity onPress={onPress}>
    <View
      style={[
        styles.iconContainer,
        isRightButton && styles.rightButtonContainer,
      ]}
    >
      {children}
    </View>
  </TouchableOpacity>
);

HeaderButton.defaultProps = {
  isRightButton: false,
};

HeaderButton.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  isRightButton: PropTypes.bool,
  onPress: PropTypes.func.isRequired,
};

export { HeaderButton };
export default withNavigation(HeaderButton);
