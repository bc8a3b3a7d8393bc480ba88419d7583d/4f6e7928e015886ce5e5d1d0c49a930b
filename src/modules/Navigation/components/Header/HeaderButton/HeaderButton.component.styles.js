import { StyleSheet } from 'react-native';
import { METRICS } from '../../../../../themes';

const styles = StyleSheet.create({
  iconContainer: {
    width: METRICS.navigation.buttonSize,
    height: METRICS.navigation.buttonSize,
    justifyContent: 'center',
    padding: METRICS.spacing.normal,
  },
  rightButtonContainer: {
    alignItems: 'flex-end',
  },
});

export default styles;
