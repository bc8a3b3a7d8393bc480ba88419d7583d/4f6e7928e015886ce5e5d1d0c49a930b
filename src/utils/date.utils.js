import moment from 'moment';

const supportedDateFormats = {
  ddddDDMMMYYYY: 'dddd, DD MMM YYYY',
};

const format = (date, dateFormat) => moment(date).format(dateFormat);

export default {
  format,
  supportedDateFormats,
};
