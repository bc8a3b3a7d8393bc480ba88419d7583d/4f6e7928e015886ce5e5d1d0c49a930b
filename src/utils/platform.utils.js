import { Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import { equals } from 'ramda';

const isDEV = () => equals(process.env.NODE_ENV, 'development');
const isIOS = () => equals(Platform.OS, 'ios');
const isAndroid = () => equals(Platform.OS, 'android');
const iPhoneX = 'iPhone X';
const isIPhoneX = () => isIOS() && equals(DeviceInfo.getModel(), iPhoneX);

export { isDEV, isIOS, isAndroid, isIPhoneX, iPhoneX };
