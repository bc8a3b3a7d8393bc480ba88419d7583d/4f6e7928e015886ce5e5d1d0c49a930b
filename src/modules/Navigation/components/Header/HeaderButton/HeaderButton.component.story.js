import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react-native';

import { HeaderButton as Component } from './HeaderButton.component';

storiesOf('Navigation/Component - HeaderButton', module).add('default', () => (
  <Component
    navigation={{
      goBack: () => console.log('go back'),
    }}
  />
));
