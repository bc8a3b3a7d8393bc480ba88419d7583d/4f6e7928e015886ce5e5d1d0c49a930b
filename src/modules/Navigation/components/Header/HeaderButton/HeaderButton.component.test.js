import React from 'react';
import { View } from 'react-native';
import { testSnapshots } from '../../../../../utils/test.utils';

import { HeaderButton } from './HeaderButton.component';

const props = {
  navigation: {
    goBack: jest.fn(),
    onPress: jest.fn(),
  },
};

describe('HeaderButton', () => {
  testSnapshots(HeaderButton, [
    {
      props: {
        ...props,
        children: <View />,
      },
      description: 'basic render',
    },
  ]);
});
