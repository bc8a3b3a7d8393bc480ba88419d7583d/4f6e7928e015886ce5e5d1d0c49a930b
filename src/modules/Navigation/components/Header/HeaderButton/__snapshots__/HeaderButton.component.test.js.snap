// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton snapshots basic render 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <Component
    style={
      Array [
        Object {
          "height": 40,
          "justifyContent": "center",
          "padding": 10,
          "width": 40,
        },
        false,
      ]
    }
  >
    <Component />
  </Component>
</TouchableOpacity>
`;

exports[`HeaderButton snapshots render right button 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <Component
    style={
      Array [
        Object {
          "height": 40,
          "justifyContent": "center",
          "padding": 10,
          "width": 40,
        },
        Object {
          "alignItems": "flex-end",
        },
      ]
    }
  >
    <Component />
  </Component>
</TouchableOpacity>
`;
